CRM/Case/Form/Activity.php

```diff
@@ -601,8 +601,16 @@ public function postProcess($params = NULL) {
             $mailStatus = ts("A copy of the activity has also been sent to selected contacts(s).");
           }
           else {
-            $this->_relatedContacts = CRM_Activity_BAO_ActivityAssignment::getAssigneeNames([$vval['actId']], TRUE, FALSE);
-            $mailStatus .= ' ' . ts("A copy of the activity has also been sent to assignee contacts(s).");
+            $actTypeId = civicrm_api3('Activity', 'getvalue', [
+              'return' => 'activity_type_id',
+              'id' => $vval['actId'],
+            ]);
+            if (!in_array($actTypeId, Civi::settings()->get('do_not_notify_assignees_for'))) {
```

Comment:
This is ok. I was thinking maybe on a site where you had never saved the display prefs page it might be null, but it always comes thru as an array even if it's not in the db settings table.

Reply:
Thanks @demeritcowboy , Have updated the changes

Reply:
Thanks. Sorry what I meant was that it was ok already because it turns out that Civi::settings()->get('do_not_notify_assignees_for') always returns an array in all situations, because it's magical. But it's ok with the extra check too - it's good to explicitly check.

Reply:
Unfortunately it returns NULL when there is no entry in civicrm_setting table, not sure why hence i added additional check to handle warnings

Reply:
Hmm, ok. My testing showed it was always an array for that setting, but anyway the check is good. Thanks.